pkg/apis/v1alpha1/subnetset_types.go

```diff
@@ -14,7 +14,10 @@ type SubnetSetSpec struct {
 	// +kubebuilder:validation:Minimum:=16
 	IPv4SubnetSize int `json:"ipv4SubnetSize,omitempty"`
 	// Access mode of Subnet, accessible only from within VPC or from outside VPC.
-	// +kubebuilder:validation:Enum=Private;Public
+	// +kubebuilder:default:=Private
+	// +kubebuilder:validation:Enum=Private;Public;PrivateTGW
+	// +kubebuilder:validation:Required
```

Comment:
Is it required if we have default type

Reply:
kubebuilder:default:=Private, this is not required, we can remove it if we don't want a default value.
but I think we want Private as default value

Reply:
I think we don't need a default value here, in the subnet controller, the `AccessMode` will be set a default value: https://github.com/vmware-tanzu/nsx-operator/blob/2cc0f8de2fc1615b517c25b5e32bebfa51aeaa71/pkg/controllers/subnet/subnet_controller.go#L81

Reply:
https://github.com/vmware-tanzu/nsx-operator/pull/664/files#diff-30b5978138c76f4191e9f5dbdc5c051570ab228c98ab270b9f7e30db1ae42d9eR66
we  will not have defaultSubnetAccessMode field any more in VPCNetworkConfiguration CR, which is being changed to podSubnetAccessMode(for POD subnetset only).
Also possibly, podSubnetAccessMode will be removed as well. 
That is to say there is no any subnet accessmode setting in VPCNetworkConfiguration CR any more. 
We can remove the default value: private from CRD definition at this moment.
If necessary, we can set default subnet/subnetset accessmode in the controller to the value we want in the future.

Reply:
remove default value for both subnet and subnetset.